vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExternalLinkManager.java

```diff
@@ -0,0 +1,118 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.vspace.core.data.ExternalLinkDisplayRepository;
+import edu.asu.diging.vspace.core.data.ExternalLinkRepository;
+import edu.asu.diging.vspace.core.exception.LinkDoesNotExistsException;
+import edu.asu.diging.vspace.core.factory.IExternalLinkDisplayFactory;
+import edu.asu.diging.vspace.core.factory.IExternalLinkFactory;
+import edu.asu.diging.vspace.core.model.IExternalLink;
+import edu.asu.diging.vspace.core.model.ILink;
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.IVSpaceElement;
+import edu.asu.diging.vspace.core.model.display.IExternalLinkDisplay;
+import edu.asu.diging.vspace.core.model.display.ILinkDisplay;
+import edu.asu.diging.vspace.core.model.display.impl.ExternalLinkDisplay;
+import edu.asu.diging.vspace.core.model.impl.ExternalLink;
+import edu.asu.diging.vspace.core.model.impl.ExternalLinkValue;
+import edu.asu.diging.vspace.core.services.IExternalLinkManager;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+
+@Transactional
+@Service
+public class ExternalLinkManager extends LinkManager implements IExternalLinkManager{
+
+    @Autowired
+    private ISpaceManager spaceManager;
+
+    @Autowired
+    private ExternalLinkRepository externalLinkRepo;
+
+    @Autowired
+    private ExternalLinkDisplayRepository externalLinkDisplayRepo;
+
+    @Autowired
+    private IExternalLinkFactory externalLinkFactory;
+
+    @Autowired
+    private IExternalLinkDisplayFactory externalLinkDisplayFactory;
+
+    @Override
+    public List<ILinkDisplay> getLinkDisplays(String spaceId) {
+        return new ArrayList<>(externalLinkDisplayRepo.findExternalLinkDisplaysForSpace(spaceId));
+    }
+
+    @Override
+    public void deleteLink(String linkId) {
```

Comment:
my suspicion is that this should be part of linkmanager too and just call abstract methods.

Reply:
we don't have much common code in the delete method which can be coded in LinkManager superclass. It would be just one abstract delete method that can be defined in sub-classes and invoked through parent delete in LinkManager.

Reply:
they all the the same thing though, you can probably write it like this in the superclass:
```
ILink link = getLinkById(linkId)
// ... return if null

removeFromLinkList(link.getSpace()) // abstract method
deleteLinkDisplay() // abstract method
deleteLink() // abstract method
```
this way the subclasses just have to implement the abstract methods and don't worry about the order. If we make changes that require the delete method to change order or do something additional, all we have to do is to change the one method in the link manager.

there is probably an even more elegant way using generics, but this should work.

Reply:
?